cilium-cli/connectivity/perf/benchmarks/netperf/perfpod.go

```diff
@@ -204,10 +199,58 @@ func NetperfCmd(ctx context.Context, sip string, perfTest common.PerfTests, a *c
 			a.Fatal("Unable to process netperf result")
 		}
 	}
-	if strings.HasSuffix(perfTest.Test, "_RR") || strings.HasSuffix(perfTest.Test, "_CRR") {
+	if strings.HasSuffix(test, "_RR") || strings.HasSuffix(test, "_CRR") {
 		// We don't want to report throughput
 		res.ThroughputMetric = nil
 	}
 
 	return res
 }
+
+func NetperfCmd(ctx context.Context, sip string, perfTest common.PerfTests, a *check.Action) common.PerfResult {
+	test := strings.TrimSuffix(perfTest.Test, "_MULTI")
+
+	streams := uint(1)
+	if strings.HasSuffix(perfTest.Test, "_MULTI") {
+		streams = perfTest.Streams
+
+		if !strings.HasSuffix(test, "_STREAM") {
+			a.Fatal("Only STREAM tests support parallelism")
+		}
+	}
+
+	args := []string{"-o", "MIN_LATENCY,MEAN_LATENCY,MAX_LATENCY,P50_LATENCY,P90_LATENCY,P99_LATENCY,TRANSACTION_RATE,THROUGHPUT,THROUGHPUT_UNITS"}
+	if test == "UDP_STREAM" || perfTest.NetQos {
+		args = append(args, "-m", fmt.Sprintf("%d", perfTest.MsgSize))
+	}
+	exec := buildExecCommand(test, sip, perfTest.Duration, args)
+
+	if streams >= 2 {
+		exec = []string{"/bin/bash", "-c",
+			// We write the output of each process to a separate file and cat them
+			// at the end to prevent the possibility of interleaved output.
+			fmt.Sprintf("DIR=$(mktemp -d); for i in {1..%d}; do %s > $DIR/out$i.out & done; wait; cat $DIR/*; rm -rf $DIR",
+				streams, strings.Join(exec, " "),
+			)}
+	}
+
+	a.ExecInPod(ctx, exec)
+	output := a.CmdOutput()
+	a.Debug("Netperf output: ", output)
+	lines := slices.DeleteFunc(
+		strings.Split(output, "\n"),
+		// Result lines always start with a number, hence drop all the others.
+		func(line string) bool { return len(line) == 0 || line[0] < '0' || line[0] > '9' },
```

Comment:
I know I am guilty of it as well, but could we add a simple unit test for parsing netperf output :pleading_face: ?

Reply:
Added in https://github.com/cilium/cilium/pull/38027/commits/d6fbdca5a27537ea347eb3aa5369d10dc304aba5.